Error while cross compiling from Windows 10 to Raspberry Pi 2 (armv7-unknown-linux-gnueabihf)

Hello, I just ran into an issue trying to use the `vendored` feature to cross-compile from Windows 10 to a Raspberry Pi 2. I am using the `nightly-x86_64-pc-windows-msvc` toolchain. The exact error can be seen in the captured stderr when the build fails: 
```
C:/SysGCC/raspberry/bin/arm-linux-gnueabihf-ar: creating apps/libapps.a
  crypto/cversion.c: In function 'OpenSSL_version':
  crypto/cversion.c:38:44: error: incomplete universal character name \U
           return "ENGINESDIR: \"" ENGINESDIR "\"";
                                              ^~~~
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\D'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\h'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\d'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\o'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\o'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\o'
  crypto/cversion.c:38:44: warning: unknown escape sequence: '\i'
  make[1]: *** [Makefile:2254: crypto/cversion.o] Error 1
  make: *** [Makefile:177: build_libs] Error 2
``` 

For my build, I specified the following environment variables:
```
PERL = "C:/Users/nathaniel/scoop/apps/msys2/current/usr/bin/perl" 

CC = "C:/SysGCC/raspberry/bin/arm-linux-gnueabihf-gcc"
CXX = "C:/SysGCC/raspberry/bin/arm-linux-gnueabihf-g++"

CMAKE_GENERATOR = "Ninja"
```

I also set the linker through `RUSTFLAGS` as `C:/SysGCC/raspberry/bin/arm-linux-gnueabihf-gcc`. I set my target as `armv7-unknown-linux-gnueabihf`.

I tracked this down to `ENGINESDIR` having backslashes instead of forward slashes, and I tracked it back to the prefix option in the configure script. Currently, this library uses the following to set the prefix:
```rust
if host.contains("pc-windows-gnu") {
    configure.arg(&format!("--prefix={}", sanitize_sh(&install_dir)));
} else {
    configure.arg(&format!("--prefix={}", install_dir.display()));
}
```

I edited the bottom case to instead be `configure.arg(&format!("--prefix={}", install_dir.to_str().unwrap().replace('\\', "/")));`. I don't particularly understand the ramifications of this change, but it seems to fix my build. `sanitize_sh` seems to only be applicable for Mingw as it tries to change the paths from Windows style to Mingw style. Could this be fixed by adding another case for windows msvc?

i did this in Cargo.toml
```
...
# openssl-sys = { path = "../openssl-sys", features = ["vendored"], version = "0.9.58" }
openssl-sys = { version = "0.9.58", features = ["vendored"]}

[patch.crates-io]
openssl-src = { path = "../openssl-src" }
openssl-sys = { path = "../openssl-sys" }
```
and still

```bash
 crypto/cversion.c:38:33: error: \U used with no following hex digits
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:23: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                        ^~
  crypto/cversion.c:38:33: warning: unknown escape sequence '\D' [-Wunknown-escape-sequence]
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:29: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                              ^~
  crypto/cversion.c:38:33: warning: unknown escape sequence '\G' [-Wunknown-escape-sequence]
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:39: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                        ^~
  crypto/cversion.c:38:33: warning: unknown escape sequence '\d' [-Wunknown-escape-sequence]
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:101: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                                                                                      ^~
  crypto/cversion.c:38:33: error: expected '{' after '\d' escape sequence
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:113: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                                                                                                  ^~
  crypto/cversion.c:38:33: error: expected '{' after '\d' escape sequence
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:142: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                                                                                                                               ^~
  crypto/cversion.c:38:33: error: expected '{' after '\d' escape sequence
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:146: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                                                                                                                                   ^~
  crypto/cversion.c:38:33: warning: unknown escape sequence '\i' [-Wunknown-escape-sequence]
          return "ENGINESDIR: \"" ENGINESDIR "\"";
                                  ^~~~~~~~~~
  <command line>:13:160: note: expanded from here
  #define ENGINESDIR "D:\Users\Documents\GitHub\aw-android\aw-server-rust\target\aarch64-linux-android\debug\build\openssl-sys-ee67b57d3d3215a1\out\openssl-build\install/lib/engines-1.1"
                                                                                                                                                                 ^~
  4 warnings and 4 errors generated.
  make[3]: *** [crypto/cversion.o] Error 1
  make[2]: *** [build_libs] Error 2
  thread 'main' panicked at '


  Error building OpenSSL:
      Command: "make" "build_libs"
      Exit status: exit code: 2


      ', C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-src-111.22.0+1.1.1q\src\lib.rs:490:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make[1]: *** [android] Error 101
make[1]: Leaving directory `D:/Users/Documents/GitHub/aw-android/aw-server-rust'
make: *** [aw-server-rust/target/armv7-linux-androideabi/debug/libaw_server.so] Error 2
```
C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-src-111.22.0+1.1.1q\src\lib.rs:490:13 - its still going somewhere else and not the ../openssl-src or ../openssl-sys

Not that good with rust building ( feels like shet :/ )

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -140,7 +140,10 @@ impl Build {
         if host.contains("pc-windows-gnu") {
             configure.arg(&format!("--prefix={}", sanitize_sh(&install_dir)));
         } else if host.contains("pc-windows-msvc") {
-            configure.arg(&format!("--prefix={}", install_dir.to_str().unwrap().replace("\\", "/")));
+            configure.arg(&format!(
+                "--prefix={}",
+                install_dir.to_str().unwrap().replace("\\", "/")
+            ));
         } else {
             configure.arg(&format!("--prefix={}", install_dir.display()));
         }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -140,6 +140,10 @@ impl Build {
         if host.contains("pc-windows-gnu") {
             configure.arg(&format!("--prefix={}", sanitize_sh(&install_dir)));
         } else if host.contains("pc-windows-msvc") {
+            // On Windows, the prefix argument does not support \ path seperators
+            // when cross compiling.
+            // Always use / as a path seperator instead of \, since that works for both
+            // native and cross builds.
             configure.arg(&format!(
                 "--prefix={}",
                 install_dir.to_str().unwrap().replace("\\", "/")

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -139,6 +139,8 @@ impl Build {
         // Change the install directory to happen inside of the build directory.
         if host.contains("pc-windows-gnu") {
             configure.arg(&format!("--prefix={}", sanitize_sh(&install_dir)));
+        } else if host.contains("pc-windows-msvc") {
+            configure.arg(&format!("--prefix={}", install_dir.to_str().unwrap().replace("\\", "/")));
         } else {
             configure.arg(&format!("--prefix={}", install_dir.display()));
         }

